sim/battle.ts

```diff
@@ -834,6 +834,14 @@ export class Battle {
 			// 	('0000' + (this.event.modifier * 4096).toString(16)).slice(-4).toUpperCase());
 			relayVar = this.modify(relayVar, this.event.modifier);
 		}
+
+		// handle haze/thawing in gen 1 here since it needs to be part of the BeforeMove event
```

Comment:
This doesn't feel like the right place for this, but I'm not familiar enough with the simulation engine to say where it should go instead, so I guess it's fine.